core/src/main/java/org/springframework/security/authorization/method/PreFilterAuthorizationMethodInterceptor.java

```diff
@@ -137,10 +147,24 @@ public Object invoke(MethodInvocation mi) throws Throwable {
 		MethodSecurityExpressionHandler expressionHandler = this.registry.getExpressionHandler();
 		EvaluationContext ctx = expressionHandler.createEvaluationContext(this::getAuthentication, mi);
 		Object filterTarget = findFilterTarget(attribute.getFilterTarget(), ctx, mi);
-		expressionHandler.filter(filterTarget, attribute.getExpression(), ctx);
+		Object filteredObject = expressionHandler.filter(filterTarget, attribute.getExpression(), ctx);
+		replaceFilterTargetIfNeeded(attribute.getFilterTarget(), ctx, filteredObject, mi);
```

Comment:
I'm a little nervous about this, given the obvious historical efforts to not replace the method parameter (which is why the collection filter clears the existing collection instead of creating a new one. @rwinch can you fill me in on the reason that was done that way?

Reply:
Believe it or not the clearing of collections happened before I was a part of the team, so I'm not sure the reasoning. I think the reason was to ensure that the same type that was passed in was used. It could also be because users expected references to the argument to be updated (which I understand to some degree, but I generally don't like mutating inputs)